lib/stsci/tools/bitmask.py

```diff
@@ -8,18 +9,419 @@
+        If input string value was prepended with '~' (or `flip_bits` was
+        set to `True`), then returned value will have its bits flipped
+        (inverse mask).
+
+    Examples
+    --------
+        >>> from stsci.tools.bitmask import interpret_bit_flags
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags(28))
+        '0000000000011100'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags('4,8,16'))
+        '0000000000011100'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags('~4,8,16'))
+        '1111111111100011'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags('~(4+8+16)'))
+        '1111111111100011'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags([4, 8, 16]))
+        '0000000000011100'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags([4, 8, 16], flip_bits=True))
+        '1111111111100011'
+
+    """
+    has_flip_bits = flip_bits is not None
+    flip_bits = bool(flip_bits)
+    allow_non_flags = False
+
+    if _is_int(bit_flags):
+        return (~int(bit_flags) if flip_bits else bit_flags)
+
+    elif bit_flags is None:
+        if has_flip_bits:
+            raise TypeError(
+                "Keyword argument 'flip_bits' must be set to 'None' when "
+                "input 'bit_flags' is None."
+            )
+        return None
+
+    elif isinstance(bit_flags, six.string_types):
+        if has_flip_bits:
+            raise TypeError(
+                "Keyword argument 'flip_bits' is not permitted for "
+                "comma-separated string lists of bit flags. Prepend '~' to "
+                "the string to indicate bit-flipping."
+            )
+
+        bit_flags = str(bit_flags).strip()
+
+        if bit_flags.upper() in ['', 'NONE', 'INDEF']:
+            return None
+
+        # check whether bitwise-NOT is present and if it is, check that it is
+        # in the first position:
+        bitflip_pos = bit_flags.find('~')
+        if bitflip_pos == 0:
+            flip_bits = True
+            bit_flags = bit_flags[1:].lstrip()
+        else:
+            if bitflip_pos > 0:
+                raise ValueError("Bitwise-NOT must precede bit flag list.")
+            flip_bits = False
+
+        # basic check for correct use of parenthesis:
+        while True:
+            nlpar = bit_flags.count('(')
+            nrpar = bit_flags.count(')')
+
+            if nlpar == 0 and nrpar == 0:
+                break
+
+            if nlpar != nrpar:
+                raise ValueError("Unbalanced parantheses in bit flag list.")
+
+            lpar_pos = bit_flags.find('(')
+            rpar_pos = bit_flags.rfind(')')
+            if lpar_pos > 0 or rpar_pos < (len(bit_flags) - 1):
+                raise ValueError("Incorrect syntax (incorrect use of "
+                                 "parenthesis) in bit flag list.")
+
+            bit_flags = bit_flags[1:-1].strip()
+
+        if ',' in bit_flags:
+            bit_flags = bit_flags.split(',')
```

Comment:
@MSeifert04 Thanks but here I think this would be unnecessary because after this big "if-elif-else" block ends, I make sure all flags can be converted to integers (``set(map(int, bit_flags))``) and so ``int`` will handle white spaces around "string flags".